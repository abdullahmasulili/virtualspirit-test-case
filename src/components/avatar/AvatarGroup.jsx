import React from "react";
import Avatar from "./Avatar";
import styles from "./AvatarGroup.module.css";

export default function AvatarGroup({ avatars, maxLength = 0, size }) {
  const isWithinMaxLength = maxLength > 0 && maxLength < avatars.length;

  return (
    <div className={styles.avatar__group} data-testid="avatar-group">
      {avatars ? (avatars.map((item, index) => {
        const shouldRenderAvatar = !isWithinMaxLength || index < maxLength;

        return shouldRenderAvatar ? (
          <Avatar
            key={index}
            name={item.name}
            image={item.image}
            size={size}
            index={index}
          />
        ) : null;
      })) : (
        <Avatar />
      )}

      {isWithinMaxLength && (
        <Avatar
          size={size}
          index={maxLength}
          name={avatars.length - maxLength}
          image={null}
        />
      )}
    </div>
  );
}
