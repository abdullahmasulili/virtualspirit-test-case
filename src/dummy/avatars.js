import { faker } from '@faker-js/faker'

const avatars = [
    {
        name: faker.name.fullName(),
        image: null
    },
    {
        name: faker.name.fullName(),
        image: faker.image.avatar()
    },
    {
        name: faker.name.fullName(),
        image: null
    },
    {
        name: faker.name.fullName(),
        image: null
    },
    {
        name: faker.name.fullName(),
        image: faker.image.avatar()
    },
    {
        name: faker.name.fullName(),
        image: null
    },
    {
        name: faker.name.fullName(),
        image: null
    },
    {
        name: faker.name.fullName(),
        image: faker.image.avatar()
    },
    {
        name: faker.name.fullName(),
        image: null
    },
    {
        name: faker.name.fullName(),
        image: null
    },
    {
        name: faker.name.fullName(),
        image: faker.image.avatar()
    },
]

export default avatars