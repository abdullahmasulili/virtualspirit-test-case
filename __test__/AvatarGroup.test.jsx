import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { it, expect, describe, test } from 'vitest'
import avatars from '../src/dummy/avatars'
import AvatarGroup from '../src/components/avatar/AvatarGroup';

describe('Avatar Group test', () => {
    test('should render properly', () => {
        render(<AvatarGroup />)

        expect(screen.getByText(/TB/i))
    })

    it('rendered with expected props', () => {
        const maxLength = 3
        const size = 'md'
        const container = render(
            <AvatarGroup avatars={avatars} maxLength={maxLength} size={size} />
        )
        const avatar = container.getAllByTestId('avatar')
        
        avatar.forEach(item => {
            expect(item.className.includes(`avatar--${size}`)).toBe(true)
        })
        expect(avatar.length).toBe(maxLength + 1)
        
    })

    it('should match the snapshot', () => {
        const tree = renderer.create(<AvatarGroup />)

        expect(tree).toMatchSnapshot()
    })
})