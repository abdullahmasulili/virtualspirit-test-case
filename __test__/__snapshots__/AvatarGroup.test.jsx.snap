// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar Group test > should match the snapshot 1`] = `
<div
  className="_avatar__group_8c19d2"
  data-testid="avatar-group"
>
  <div
    className="_avatar_69744d _avatar--md_69744d"
    data-testid="avatar"
    style={
      {
        "backgroundColor": "#fedca4",
        "color": "white",
        "fontSize": "12px",
        "left": "0px",
      }
    }
  >
    <span>
      TB
    </span>
  </div>
</div>
`;
